### DISCUSSION TITLE
Configurable RPC url

### DISCUSSION THREAD

**Discussion by BriungRi** [celo-org/react-celo#131] [Ideas]
Right now, the RPC defaults to forno for Celo networks: https://github.com/celo-org/use-contractkit/blob/fa397e894fa32afb0a24a0cc651f12d20d92cddc/packages/use-contractkit/src/constants.tsx#L177

I think it is important to allow this to be configurable for a few reasons:
1. Projects like Ubeswap can move off public infra and use their own node
2. If we support fallbacks, this could be a great way to add resilience if a node goes down
3. If this project can support networks outside of Celo like Ethereum and Polygon (thinking Optics powered apps for example), there are no public nodes for those networks

**Comment by aaronmgdr** [comment]
For 1. the `ContractKitProvider` takes a `networks` prop as an array of Network where each has properties for `name`, `rpcURL`, `graphQl` `explorer`, `chainId`.  Is this not working? 


2. Fallbacks is a great idea. 

3. I believe the ability to add networks as described in 1 fixes this, but i suppose there  might be some other issues to address

**Reply by BriungRi** [reply]
I didn't realize that was an option. Will certainly try this out as Ubeswap may potentially decide to move off Forno.

**Comment by BriungRi** [comment]
Possible bug: https://github.com/celo-org/use-contractkit/blob/2dfe34e51f171530c3dd3256b588d44c76c072cb/packages/use-contractkit/src/utils/helpers.ts#L67

This line overrides a custom Celo network and will just use forno unless a developer does a type-unsafe change of `[[Network]].name`

**Comment by aaronmgdr** [comment]
Networks will have no restrictions in name follow next release https://github.com/celo-org/use-contractkit/pull/168
